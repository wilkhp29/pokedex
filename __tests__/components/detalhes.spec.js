import React from 'react';
import { shallow } from 'enzyme';
import Pokemon from '../../src/Components/pokemon';
import {initalState as state} from "../mock";


jest.mock('react-redux', () => {
  const {initalState} = require("../mock");
  
   return {
        useSelector: jest.fn(fn => fn(initalState)),
        useDispatch: () => {}
    }
});


describe('pokedex', () => {
    it('renderizando  componente detalhes', () => {
        const wrapper = shallow(
            <Pokemon pokemon={state.pokemon.pokemon} />
        );
    
        expect(wrapper).toMatchSnapshot();
        wrapper.find("span").equals("25 - pikachu");
    });
})
