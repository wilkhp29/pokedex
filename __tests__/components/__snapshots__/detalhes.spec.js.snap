// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pokedex renderizando  componente detalhes 1`] = `
<styled.div
  onClick={[Function]}
>
  <header>
    <span>
      25
       - 
      pikachu
    </span>
    <img
      src="https://www.serebii.net/pokedex-sm/icon/025.png"
    />
  </header>
  <div>
    <img
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/25.png"
    />
  </div>
</styled.div>
`;
